.github/workflows/image.yaml

```diff
@@ -9,41 +9,47 @@ on:
       - master
     types: [ labeled, unlabeled, opened, synchronize, reopened ]
 
+env:
+  GOLANG_VERSION: '1.17.6'
+
 jobs:
   publish:
+    if: github.repository == 'argoproj/applicationset'
     runs-on: ubuntu-latest
     env:
       GOPATH: /home/runner/work/applicationset
     steps:
-      - uses: actions/setup-go@v1
+      - name: Setup Golang
+        uses: actions/setup-go@v1
         with:
-          go-version: '1.17.6'
-      - uses: actions/checkout@master
+          go-version: ${{ env.GOLANG_VERSION }}
+
+      - name: Checkout code
+        uses: actions/checkout@master
         with:
           path: src/applicationset
 
-      # Build the image
+      - name: Login to Quay.io container registry
+        uses: docker/login-action@v1
+        with:
+          registry: quay.io
+          username: ${{ secrets.USERNAME }}
+          password: ${{ secrets.TOKEN }}
+        if: ${{ github.event_name == 'push' }}
+
       - uses: docker/setup-qemu-action@v1
       - uses: docker/setup-buildx-action@v1
-      - run: |
+      - name: Build and publish the latest image
+        run: |
           IMAGE_PLATFORMS=linux/amd64
           if [[ "${{ github.event_name }}" == "push" || "${{ contains(github.event.pull_request.labels.*.name, 'test-arm-image') }}" == "true" ]]
           then
             IMAGE_PLATFORMS=linux/amd64,linux/arm64
           fi
           echo "Building image for platforms: $IMAGE_PLATFORMS"
-          make image DOCKER_PUSH=false CONTAINER_REGISTRY=quay.io IMAGE_NAMESPACE=argoproj IMAGE_TAG=latest IMAGE_PLATFORMS=${IMAGE_PLATFORMS}
+          make multi-images DOCKER_PUSH="${{ github.event_name == 'push' }}" \
```

Comment:
One option would be to hard-code the buildx command, mimicking what's done in Argo CD. https://github.com/argoproj/argo-cd/blob/7cf6a5b1907b8b16b57e419a52b4c34ed8c74a55/.github/workflows/image.yaml#L49

But I'm alright with the new Make target too.

Reply:
I'm fine with the hardcoded build command,  but was not sure if the phony target was needed.  Let me know what path you would like to take.   Whatever makes the devs life easier.

Reply:
Sounds great,  thanks for all your help.